drivers/spi/spi_nrfx_spim.c

```diff
@@ -416,7 +524,10 @@ static int spim_nrfx_pm_control(const struct device *dev,
 			.orc       = CONFIG_SPI_##idx##_NRF_ORC,	       \
 			.miso_pull = SPIM_NRFX_MISO_PULL(idx),		       \
 			SPI_NRFX_SPIM_EXTENDED_CONFIG(idx)		       \
-		}							       \
+		},							       \
+		COND_CODE_1(CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58,     \
+		(.anomaly_58_workaround = SPIM_PROP(idx, anomaly_58_workaround),\
+		), ())							       \
```

Comment:
Perhaps putting it this way could make it a bit more readable:
```suggestion
		COND_CODE_1(CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58,     \
			(.anomaly_58_workaround =			       \
				SPIM_PROP(idx, anomaly_58_workaround),),       \
			())						       \
```